src/juicebox/tools/utils/original/Preprocessor.java

```diff
@@ -627,10 +698,36 @@ private void writeBody(String inputFile) throws IOException {
                     continue;
                 }
 
+                // Randomize
                 // Randomize
                 if (fragmentCalculation != null && allowPositionsRandomization) {
-                    bp1 = randomizePos(fragmentCalculation, chromosomeHandler.getChromosomeFromIndex(chr1).getName(), frag1);
-                    bp2 = randomizePos(fragmentCalculation, chromosomeHandler.getChromosomeFromIndex(chr2).getName(), frag2);
+                    FragmentCalculation fragMapToUse;
+                    if (randomizeFragMaps != null) {
+                        FragmentCalculation fragMap1 = findFragMap(randomizeFragMaps, chromosomeHandler.getChromosomeFromIndex(chr1).getName(), bp1, frag1);
+                        FragmentCalculation fragMap2 = findFragMap(randomizeFragMaps, chromosomeHandler.getChromosomeFromIndex(chr2).getName(), bp2, frag2);
+
+                        if (fragMap1 == null && fragMap2 == null) {
+                            noMapFoundCount += 1;
+                            continue;
+                        } else if (fragMap1 != null && fragMap2 != null && fragMap1 != fragMap2) {
+                            mapDifferentCount += 1;
+                            continue;
+                        }
+
+                        if (fragMap1 != null) {
+                            fragMapToUse = fragMap1;
+                        } else {
+                            fragMapToUse = fragMap2;
+                        }
+
+                    } else {
+                        // use default map
+                        fragMapToUse = fragmentCalculation;
+                    }
+
+
+                    bp1 = randomizePos(fragMapToUse, chromosomeHandler.getChromosomeFromIndex(chr1).getName(), frag1);
```

Comment:
can randomizeFragMaps be renamed? I don't know exactly what it is in this context.